### PR TITLE
chore: Rename Middleware types

### PR DESCRIPTION
**Pull request by FrankEssenberger**
While adjusting the middleware documentation I found that our names are not optimal:
- The `MiddleWareIn` and `MiddlewareOut` was not super nice -> Options and Function are better names
- For the public API we should use the `HTTPRequest` type instead of the axios one.

### CODE REVIEW

File: packages/resilience/src/middleware.ts [hunk 1]
```diff
@@ -75,8 +75,8 @@ export type Middleware<
   ReturnT,
   ContextT extends MiddlewareContext<ArgumentT>
 > = (
-  options: MiddlewareIn<ArgumentT, ReturnT, ContextT>
-) => MiddlewareOut<ArgumentT, ReturnT>;
+  options: MiddlewareOptions<ArgumentT, ReturnT, ContextT>
```
Comment (deekshas8) [inline packages/resilience/src/middleware.ts:78]: [req] Update the jsdoc. It still mentions `MiddlewareIn`

File: packages/resilience/src/middleware.ts [hunk 2]
```diff
@@ -130,7 +130,7 @@ function addMiddlewaresToInitialFunction<
   ContextT extends MiddlewareContext<ArgumentT>
 >(
   middlewares: Middleware<ArgumentT, ReturnT, ContextT>[],
-  initial: MiddlewareIn<ArgumentT, ReturnT, ContextT>
+  initial: MiddlewareOptions<ArgumentT, ReturnT, ContextT>
 ): MiddlewareOut<ArgumentT, ReturnT> {
```
Comment (deekshas8) [inline packages/resilience/src/middleware.ts:134]: [q] the return type here can also be changed to `MiddlewareFunction` right?

File: packages/resilience/src/retry.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export function retry<
   }
 
   return function (
-    options: MiddlewareIn<ArgumentType, ReturnType, ContextType>
+    options: MiddlewareOptions<ArgumentType, ReturnType, ContextType>
   ): MiddlewareOut<ArgumentType, ReturnType> {
```
Comment (deekshas8) [inline packages/resilience/src/retry.ts:35]: [q] The return type here as well?

File: packages/resilience/src/middleware.ts (hunk 1 above)
Comment (FrankEssenberger) [reply packages/resilience/src/middleware.ts:78]: see here: https://github.com/SAP/cloud-sdk/pull/1234

File: packages/resilience/src/middleware.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/resilience/src/middleware.ts:134]: yes removed them all

File: packages/resilience/src/retry.ts (hunk 3 above)
Comment (FrankEssenberger) [reply packages/resilience/src/retry.ts:35]: see above

File: packages/resilience/src/middleware.ts (hunk 1 above)
Comment (deekshas8) [reply packages/resilience/src/middleware.ts:70]: No, I meant the comment in line 62. 
 

> The input is the MiddlewareIn containing the initial function and some context information e.g. axios request and the request context.

File: packages/http-client/src/http-client-types.ts [hunk 4]
```diff
@@ -1,13 +1,12 @@
 import * as http from 'http';
 import type { MiddlewareContext, Middleware } from '@sap-cloud-sdk/resilience';
 // eslint-disable-next-line import/named
```
Comment (deekshas8) [inline packages/http-client/src/http-client-types.ts:3]:
```suggestion
```

File: packages/http-client/src/http-client.spec.ts [hunk 5]
```diff
@@ -239,7 +239,7 @@ describe('generic http client', () => {
       // Doing the dummy object the name of the key is taken as function name.
       const dummy = {
         [appendedText](
-          options: MiddlewareIn<
+          options: MiddlewareOptions<
             RawAxiosRequestConfig,
```
Comment (deekshas8) [inline packages/http-client/src/http-client.spec.ts:243]: [req] change to `HttpRequestConfig`

**Review by deekshas8 — APPROVED** [review]
Left one suggestion, but otherwise, LGTM.

**Pull request merged**